src/jquery.data-remote.js

```diff
@@ -177,7 +196,7 @@
         var method = _options.oneAndDone ? 'one' : 'on';
 
         // bind to specific event type
-        $element[method](_options.eventType, function(evt) {
+        $element[method](_options.eventType, debounce(function(evt) {
```

Comment:
Good question :). I'm open to suggestions. Maybe pull out the callback function and declare it, rather than passing it anonymously. Then update it with a debounce for desired event types.

```js
var callback = function callback(evt) {
  evt.preventDefault();

  // if you're watching on keyup or change events, let's assume you want to
  // send the value of the element as a query parameter.
  // Think autosuggest search boxes.
  // <input data-event-type="keyup" name="q" data-target="#search-results">
  if (['keyup', 'change'].indexOf(evt.type) !== -1) {
    _options.data[this.name] = this.value;
  }

  // execute before request callback
  _options.before.call($element, $target);

  fetch(_options.url, {
    element: $element,
    target: $target,
    options: _options,
  });
}

if (debouncableEvents.indexOf(_options.eventType) !== -1) {
  callback = debounce(callback, 500);
}

$element[method](_options.eventType, callback);
```

I would declare `debouncableEvents` outside of that scope though. maybe below line 34 where the options are set

```js
// put a comment here please
var debouncableEvents = ['scroll', 'resize', 'keyup', 'keydown', 'keypress'];
```

Feel free to change up variable and method names to something you like better.